app/Controllers/Http/EstablishmentController.js

```diff
@@ -0,0 +1,57 @@
+'use strict'
+
+/* istanbul ignore next */
+// eslint-disable-next-line
+const Establishment = use('App/Models/Establishment')
+
+/* istanbul ignore next */
+class EstablishmentController {
+  async create({ request, response }) {
+    const establishment = await Establishment.create({
+      name: request.input('name'),
+      code: request.input('code'),
+      region_id: request.input('region_id')
+    })
+
+    return response.status(201).json({ establishment })
+  }
+
+  async read({ params, response }) {
+    let results = null
+
+    if (params.id !== undefined) {
+      results = await Establishment.find(params.id)
+    } else {
+      // TODO: Debug, not returning anything
+      results = await Establishment.all()
+    }
+    return response.status(200).json({ results })
+  }
```

Comment:
Should be split in 2 separate methods named "index" and "show" to match adonis naming convention

Reply:
Resolved by 52d5a27